pytorch_lightning/plugins/environments/lsf_environment.py

```diff
@@ -51,112 +54,152 @@ def __init__(self):
             )
         self._main_address = self._get_main_address()
         self._main_port = self._get_main_port()
-        log.debug(f"MASTER_ADDR: {self._main_address}")
-        log.debug(f"MASTER_PORT: {self._main_port}")
-
-    @property
-    def creates_processes_externally(self) -> bool:
-        return True
-
-    @property
-    def main_address(self) -> str:
-        """The main address is read from a list of hosts contained in the environment variable `LSB_HOSTS`."""
-        return self._main_address
-
-    @property
-    def main_port(self) -> int:
-        """The main port gets calculated from the LSF job ID."""
-        return self._main_port
+        self._local_rank = self._get_local_rank()
+        self._global_rank = self._get_global_rank()
+        self._world_size = self._get_world_size()
+        self._node_rank = self._get_node_rank()
+
+        # set environment variables needed for initializing torch distributed process group
+        os.environ["MASTER_ADDR"] = str(self._main_address)
+        log.debug(f"MASTER_ADDR: {os.environ['MASTER_ADDR']}")
+        os.environ["MASTER_PORT"] = str(self._main_port)
+        log.debug(f"MASTER_PORT: {os.environ['MASTER_PORT']}")
+
+        tmp = ("main_address", "main_port", "world_size", "local_rank", "node_rank", "global_rank")
+        self._rep = ",".join("{}={}".format(s, getattr(self, "_" + s)) for s in tmp)
+
+    def _read_hosts(self):
```

Comment:
@ajtritt You have changed and re-arranged several methods in this class. What is the reason for this? It would be much more practical for the review process to focus on one change alone, the parsing of LSB_DJOB_RANKFILE.

Reply:
@awaelchli I was asked to share my version of the code that works with the LSF environment on Summit. I copied over the code that I use and have verified works, and then adapted to work with the new state of the API.

Reply:
Ok, no problem. I just didn't expect there to be this many changes. 
I will see if I can re-order the methods to reduce the diff.

Reply:
I can do that when I address suggestions/comments from @tchaton